docs/content/develop/resource.md

```diff
@@ -0,0 +1,455 @@
+   # provider.yaml.
+   # min_version: beta
+
+   # Inserts styled markdown into the header of the resource's page in the
+   # provider documentation.
+   # docs:
+   #   warning: WARNING_MARKDOWN
+   #   note: NOTE_MARKDOWN
+
+   # URL for the resource within the API domain. This should match the
+   # resource's create URL (excluding any query parameters).
+   # Terraform field names enclosed in double curly braces will be replaced
+   # with the field values from the resource.
+   base_url: 'projects/{{project}}/locations/{{location}}/resourcenames'
+   # URL for a created resource within the API domain. This should match
+   # the URL for getting a single resource.
+   # Terraform field names enclosed in double curly braces will be replaced
+   # with the field values from the resource.
+   self_link: 'projects/{{project}}/locations/{{location}}/resourcenames/{{resource_id}}'
+   # URL for importing a resource that already exists in GCP. In general
+   # this will be a list containing self_link. If the resource cannot be read
+   # from GCP, comment this out and set exclude_import: true instead.
+   import_format: ['projects/{{project}}/locations/{{location}}/resourcenames/{{resource_id}}']
+   # If true, import code will not be generated for this resource. This should
+   # only be used if the resource cannot be read from GCP due to either API or
+   # provider constraints.
+   # exclude_import: true
+
+   # If true, the resource and all its fields will be considered immutable - that
+   # is, only creatable, not updatable. Individual fields can override this if
+   # they have a custom update method in the API.
+   # immutable: true
+
+   # Overrides one or more timeouts, in minutes. All timeouts default to 20.
+   # timeouts: !ruby/object:Api::Timeouts
+   #   insert_minutes: 20 
+   #   update_minutes: 20 
+   #   delete_minutes: 20 
+
+   # URL for creating a new resource, including query parameters.
+   # Terraform field names enclosed in double curly braces will be replaced
+   # with the field values from the resource.
+   create_url: 'projects/{{project}}/locations/{{location}}/resourcenames?resourceId={{resource_id}}'
+   # Overrides the HTTP verb used to create a new resource.
+   # Allowed values: :POST, :PUT, :PATCH. Default: :POST
+   # create_verb: :POST
+
+   # Overrides the update URL for the resource. (Otherwise, the self_link URL
+   # will be used.)
+   # Terraform field names enclosed in double curly braces will be replaced
+   # with the field values from the resource.
+   # update_url: 'projects/{{project}}/locations/{{location}}/resourcenames/{{resource_id}}'
+   # The HTTP verb used to update a resource. Allowed values: :POST, :PUT, :PATCH. Default: :PUT.
+   update_verb: :PATCH
+   # True if the resource should use an update mask for updates.
+   update_mask: true
+
+   # Overrides the delete URL for the resource. (Otherwise, the self_link URL
```

Comment:
Not sure why we're inconsistent about defaults & guidance, the same is true of the other URLs.

Reply:
I'm not sure what you mean, could you elaborate?

Reply:
Create is "URL for the resource's standard Create method" while this and update are overrides

Reply:
Create generally _has_ to be specified because it can't be inferred from the base url, whereas delete and update can be. It seems like this isn't a blocker so I'll leave it as is for now but I'm happy to change later if you have concerns!